csharp/src/Ice/Communicator.cs

```diff
@@ -793,45 +794,38 @@ public void AddAdminFacet(string facet, IObject servant)
+        /// with the locator registry. CreateAdmin can only be called once; subsequent calls throw
+        /// <see cref="InvalidOperationException"/>.</summary>
+        /// <param name="adminAdapter">The object adapter used to host the Admin object; if null and Ice.Admin.Endpoints
+        /// is set, create, activate and use the Ice.Admin object adapter.</param>
         /// <param name="adminIdentity">The identity of the Admin object.</param>
+        /// <param name="cancel">The cancellation token.</param>
         /// <returns>A proxy to the Admin object.</returns>
-        public IObjectPrx CreateAdmin(ObjectAdapter? adminAdapter, Identity adminIdentity)
-        {
-            try
-            {
-                ValueTask<IObjectPrx> task = CreateAdminAsync(adminAdapter, adminIdentity);
-                return task.IsCompleted ? task.Result : task.AsTask().Result;
-            }
-            catch (AggregateException ex)
-            {
-                Debug.Assert(ex.InnerException != null);
-                throw ExceptionUtil.Throw(ex.InnerException);
-            }
-        }
+        public IObjectPrx CreateAdmin(
+            ObjectAdapter? adminAdapter,
+            Identity adminIdentity,
+            CancellationToken cancel = default) =>
+            CreateAdminAsync(adminAdapter, adminIdentity, cancel).GetResult();
 
-        /// <summary>Add the Admin object with all its facets to the provided object adapter. If Ice.Admin.ServerId is
-        /// set and the provided object adapter has a Locator, createAdmin registers the Admin's Process facet with the
-        /// Locator's LocatorRegistry. CreateAdmin can only be called once; subsequent calls raise
-        /// InvalidOperationException.</summary>
-        /// <param name="adminAdapter">The object adapter used to host the Admin object; if null and
-        /// Ice.Admin.Endpoints is set, create, activate and use the Ice.Admin object adapter.</param>
+        /// <summary>Adds the Admin object with all its facets to the provided object adapter. If Ice.Admin.ServerId is
+        /// set and the provided object adapter has a locator registry, CreateAdmin registers the Admin's Process facet
+        /// with the locator registry. CreateAdmin can only be called once; subsequent calls throw
+        /// <see cref="InvalidOperationException"/>.</summary>
+        /// <param name="adminAdapter">The object adapter used to host the Admin object; if null and Ice.Admin.Endpoints
+        /// is set, create, activate and use the Ice.Admin object adapter.</param>
         /// <param name="adminIdentity">The identity of the Admin object.</param>
+        /// <param name="cancel">The cancellation token.</param>
         /// <returns>A proxy to the Admin object.</returns>
-        public async ValueTask<IObjectPrx> CreateAdminAsync(ObjectAdapter? adminAdapter, Identity adminIdentity)
+        public async ValueTask<IObjectPrx> CreateAdminAsync(
```

Comment:
Shouldn't we use `Task` instead of `ValueTask` for public methods?

Reply:
I believe the rule is or should be "return a ValueTask when synchronous completion is common". Here it's fairly common: the private SetServerProcessProxyAsync returns a ValueTask for the common situation where it does nothing (when there is no locator registry to call).